server/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -474,6 +479,9 @@ public void writeTo(StreamOutput out) throws IOException {
         if (out.getTransportVersion().onOrAfter(TransportVersions.INGEST_REQUEST_INCLUDE_SOURCE_ON_ERROR)) {
             out.writeBoolean(includeSourceOnError);
         }
+        if (out.getTransportVersion().onOrAfter(TransportVersions.STREAMS_ENDPOINT_PARAM_RESTRICTIONS)) {
+            out.writeCollection(paramsUsed, StreamOutput::writeString);
```

Comment:
I'm surprised there's no BulkRequest serialization test that you could add to for this change. This change is simple enough, but might be worth adding a roundtrip serialization test.